presto-main/src/main/java/com/facebook/presto/operator/scalar/StringFunctions.java

```diff
@@ -91,22 +93,78 @@ public static Slice replace(@SqlType(StandardTypes.VARCHAR) Slice str, @SqlType(
     @SqlType(StandardTypes.VARCHAR)
     public static Slice replace(@SqlType(StandardTypes.VARCHAR) Slice str, @SqlType(StandardTypes.VARCHAR) Slice search, @SqlType(StandardTypes.VARCHAR) Slice replace)
     {
-        String replaced = str.toString(UTF_8).replace(
-                search.toString(UTF_8),
-                replace.toString(UTF_8));
-        return Slices.copiedBuffer(replaced, UTF_8);
+        // Empty search?
+        if (search.length() == 0) {
+            // With empty `search` we insert `replace` in front of every character and and the end
+            Slice buffer = Slices.allocate((countCodePoints(str) + 1) * replace.length() + str.length());
```

Comment:
I meant `Slices.allocate((str.length() + 1) * replace.length() + str.length())`. Number of code points is irrelevant here.